apps/rush/src/actions/RebuildAction.ts

```diff
@@ -148,17 +148,32 @@ export default class RebuildAction extends CommandLineAction {
       () => {
         stopwatch.stop();
         console.log(colors.green(`rush ${this.options.actionVerb} (${stopwatch.toString()})`));
+        this._collectTelemetry(stopwatch, true);
       },
       () => {
         stopwatch.stop();
         console.log(colors.red(`rush ${this.options.actionVerb} - Errors! (${stopwatch.toString()})`));
-        process.exit(1);
+        this._collectTelemetry(stopwatch, false);
+        this._parser.exitWithError();
       })
       .then(() => {
         this._eventHooksManager.handle(Event.postRushBuild);
       });
   }
 
+  private _collectTelemetry(stopwatch: Stopwatch, success: boolean): void {
+    this._parser.telemetry.log({
```

Comment:
for example exitWithError

---
In reply to: [117844885](https://github.com/Microsoft/web-build-tools/pull/207#discussion_r117844885) [](ancestors = 117844885)